command/agent/cache_end_to_end_test.go

```diff
@@ -482,5 +474,67 @@ func TestCache_ClientAutoAuth(t *testing.T) {
 			t.Fatal(err)
 		}
 	}
+}
+
+func TestCache_ClientChooseSinks(t *testing.T) {
+	logger := logging.NewVaultLogger(log.Trace)
+	cluster := testAutoAuthCluster(t)
+	defer cluster.Cleanup()
+
+	tokenTtl := 200 * time.Millisecond
+
+	// Test that multiple sinks returns an error when no AgentSinkName given
+	{
+		sink1 := newFileSink(t, logger)
+		sink1.Name = "sink1"
+		out := sink1.Config["path"].(string)
+
+		sink2 := newFileSink(t, logger)
+		sink2.Name = "sink2"
+
+		cleanup, _, agentAddr := testhelperAutoAuth(t, cluster, []*sink.SinkConfig{sink1, sink2}, out, tokenTtl)
+
+		clientConfig := api.DefaultConfig()
+		clientConfig.AgentAddress = agentAddr
+		_, err := api.NewClient(clientConfig)
+		if err == nil {
+			t.Fatalf("expected error: multiple published sinks but no name or filepath given")
+		}
+
+		cleanup()
+	}
+
+	// Test that with multiple sinks we choose the right one when AgentSinkName given
+	{
+		sink1 := newFileSink(t, logger)
+		sink1.Name = "sink1"
+		out := sink1.Config["path"].(string)
+
+		sink2 := newFileSink(t, logger)
+		sink2.Name = "sink2"
+
+		cleanup, _, agentAddr := testhelperAutoAuth(t, cluster, []*sink.SinkConfig{sink1, sink2}, out, tokenTtl)
+		os.Remove(out)
+
+		clientConfig := api.DefaultConfig()
+		clientConfig.AgentAddress = agentAddr
+
+		// TODO uncomment once AgetnSinkName is supported
+		// clientConfig.AgentSinkName = "sink2"
+
+		client, err := api.NewClient(clientConfig)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = client.Logical().Read("kv/foo")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cleanup()
+	}
+}
 
+func TestCache_ClientAutoAuthEnc(t *testing.T) {
```

Comment:
Can we remove this function?

Reply:
It's there as a stub to remind me that there's a test needed for that functionality.